ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -240,7 +240,7 @@ public SchemaKStream selectKey(final Field newKeyField, boolean updateRowKey) {
     }
 
 
-    KStream keyedKStream = kstream.selectKey((key, value) -> {
+    KStream keyedKStream = kstream.filter((key, value) -> value != null).selectKey((key, value) -> {
```

Comment:
While we're at it should we fix the NPE that I think would get thrown on line 248 if the value in the key column is null?

Reply:
Good catch @rodesai , I agree that it would be nice to fix it as part of this patch.

Reply:
sure. will do